backend/controller/admin/admin.go

```diff
@@ -207,3 +225,50 @@ func (s *AdminService) SecretUnset(ctx context.Context, req *connect.Request[ftl
 	}
 	return connect.NewResponse(&ftlv1.UnsetSecretResponse{}), nil
 }
+
+func (s *AdminService) validateAgainstSchema(ctx context.Context, isSecret bool, ref cf.Ref, value json.RawMessage) error {
+	sch, err := s.schr.GetActiveSchema(ctx)
+	if err != nil {
+		return fmt.Errorf("could not get active schema: %w", err)
+	}
+
+	r := schema.RefKey{Module: ref.Module.Default(""), Name: ref.Name}.ToRef()
+	decl, ok := sch.Resolve(r).Get()
+	if !ok {
+		// Globals aren't in the module schemas, so we have nothing to validate them against.
+		if !ref.Module.Ok() {
+			return nil
+		}
+		return fmt.Errorf("declaration %q not found", ref.Name)
+	}
+
+	var fieldType schema.Type
+	if isSecret {
+		decl, ok := decl.(*schema.Secret)
+		if !ok {
+			return fmt.Errorf("%q is not a secret declaration", ref.Name)
+		}
+		fieldType = decl.Type
+	} else {
+		decl, ok := decl.(*schema.Config)
+		if !ok {
+			return fmt.Errorf("%q is not a config declaration", ref.Name)
+		}
+		fieldType = decl.Type
+	}
+
+	var v any
+	dec := json.NewDecoder(bytes.NewReader(value))
```

Comment:
This will have to use `go-runtime/encoding` - which we still need to pull out of the runtime.